kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -249,10 +262,20 @@ impl<'tcx> GotocCtx<'tcx> {
                 let (tmp, decl_stmt) =
                     self.decl_temp_variable(var1.typ().clone(), Some(var1.to_owned()), loc);
                 let var2 = fargs.remove(0);
-                let op_expr = (var1.clone()).$op(var2).with_location(loc);
-                let assign_stmt = (var1.clone()).assign(op_expr, loc);
-                let res_stmt = self.codegen_expr_to_place_stable(place, tmp.clone());
-                Stmt::atomic_block(vec![decl_stmt, assign_stmt, res_stmt], loc)
+                if var2.typ().is_pointer() {
+                    let op_expr = (var1.clone().cast_to(Type::c_size_t()))
```

Comment:
Only `op_expr` depends on `var2` being a pointer or not. The rest of code can be out of the if-then-else blocks.